File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.py
def solution(lines):
    set1 = set(i for i in range(lines[0][0], lines[0][1]))
    set2 = set(i for i in range(lines[1][0], lines[1][1]))
    set3 = set(i for i in range(lines[2][0], lines[2][1]))
    return len((set1 & set2) | (set2 & set3) | (set1 & set3))

File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.js
function solution(lines) {
    let min = Math.min(...lines.flat());
    let max = Math.max(...lines.flat());
    let numArr = Array(max-min+1).fill(0);

    for (let line of lines) {
        if (min < 0) {
            line[0] -= min
            line[1] -= min
        }
        for (let i = line[0]; i < line[1]; i++) numArr[i]++;
    }

    return numArr.filter(v => v > 1).length;
}